### PR TITLE
Examine Overhaul

### PR DESCRIPTION
**Pull request by Geevies**
image
image
image
image

### CODE REVIEW

**Comment by Geevies** [comment]
!review

**Comment by alsoandanswer** [comment]
i'm pretty sure mwaha's working on this

https://github.com/Aurorastation/Aurora.3/pull/7217

maybe you can take the reigns from him?

**Comment by Geevies** [comment]
Ours does different things really. But I can steal some features from that PR, yeah.

**Comment by Geevies** [comment]
image

File: code/modules/heavy_vehicle/mecha.dm [hunk 1]
```diff
@@ -93,9 +93,17 @@
 	if(!user || !user.client)
 		return
 	to_chat(user, "That's \a [src].")
-	user << desc
+	to_chat(user, desc)
 	if(LAZYLEN(pilots) && (!hatch_closed || body.pilot_coverage < 100 || body.transparent_cabin))
-		to_chat(user, "It is being piloted by [english_list(pilots, nothing_text = "nobody")].")
+		if(length(pilots) == 0)
+			to_chat(user, "It is being piloted by nobody.")
```
Comment (NonQueueingMatt) [inline code/modules/heavy_vehicle/mecha.dm]: This is a really odd sentence.

File: code/modules/heavy_vehicle/mecha.dm [hunk 2]
```diff
@@ -93,9 +93,17 @@
...
+		else
+			for(var/pilot in pilots)
+				if(istype(pilot, /mob/))
```
Comment (NonQueueingMatt) [inline code/modules/heavy_vehicle/mecha.dm]:
```suggestion
				if(istype(pilot, /mob))
```

File: code/modules/heavy_vehicle/mecha.dm (hunk 1 above)
Comment (Geevies) [inline code/modules/heavy_vehicle/mecha.dm]:
```suggestion
			to_chat(user, "It has no pilot.")
```

**Comment by KasparoVy** [comment]
Massive undertaking but big payout!

File: code/game/atoms.dm [hunk 3]
```diff
@@ -146,9 +146,18 @@
...
 	to_chat(user, "\icon[src] That's [f_name] [suffix]")
 	to_chat(user, desc)
+	if(description_fluff)
+		to_chat(user, span("notice", "This item has additional examine info. <a href=?src=\ref[src];examine=fluff>\[View\]</a>"))
+	if(description_antag && player_is_antag(user.mind))
+		to_chat(user, span("notice", "This item has additional antag info. <a href=?src=\ref[src];examine=fluff>\[View\]</a>"))
 
 	return distance == -1 || (get_dist(src, user) <= distance)
 
+/atom/Topic(href,href_list[])
```
Comment (skull132) [inline code/game/atoms.dm]:
```suggestion
/atom/Topic(href,href_list[])
	. = ..()
	if (.)
		return

```

**Pull request merged**